### PR TITLE
feat: Add `SwapLite` util functions

### PR DESCRIPTION
**Pull request by abcrane123**
**What changed? Why?**
- `useResetSwapLiteInputs` : resets to token, from token, USDC, and ETH balances 
- `getSwapLiteQuote`: formats swap quote values
- `useSwapLiteTokens`: returns token information (balance, etc)  - for context, USDC and ETH will always be provided as swap options. users then have the ability to provide one more `from` token to the component

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: src/core/api/getSwapLiteQuote.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+type GetSwapLiteQuoteParams = Omit<GetSwapQuoteParams, 'from'> & {
+  fromSwapUnit: SwapUnit;
+  from?: Token;
+};
+
+/**
+ * Fetches a quote for a swap, but only if the from and to tokens are different.
+ */
+
+export async function getSwapLiteQuote({
```
Comment (dschlabach) [inline src/core/api/getSwapLiteQuote.ts:27]: Why not just check if to === from and then return early if that's the case?

Comment (abcrane123) [reply src/core/api/getSwapLiteQuote.ts:27]: we're fetching three quotes (to <> ETH, to <> USDC, to <> customToken) so we wouldn't want to return early. we want to gracefully handle case where any of those tokens overlap.

File: src/core/api/getSwapLiteQuote.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  let formattedFromAmount = '';
+  if (response && !isSwapError(response)) {
+    formattedFromAmount = response?.fromAmount
+      ? formatTokenAmount(response.fromAmount, response.from.decimals)
+      : '';
+
+    fromSwapUnit.setAmountUSD(response?.fromAmountUSD || '');
+    fromSwapUnit.setAmount(formattedFromAmount || '');
```
Comment (alessey) [inline src/core/api/getSwapLiteQuote.ts:61]: if this is going to be a core/api, it shouldn't setState, is there a reason a consumer would use this vs. getSwapQuote?

File: src/core/api/getSwapLiteQuote.ts (hunk 1 above)
Comment (abcrane123) [reply src/core/api/getSwapLiteQuote.ts:27]: you can checkout the implementation of this hook [here](https://github.com/coinbase/onchainkit/pull/1729/files#diff-5fdf01a2dda1a451f5038f451d0d90a5ccbd38f2e31835a0470e631049811cc6R273)

File: src/core/api/getSwapLiteQuote.ts (hunk 2 above)
Comment (abcrane123) [reply src/core/api/getSwapLiteQuote.ts:61]: oh yeah good catch let me move it to `swap`. no reason to have this here.

File: src/swap/hooks/useSwapLiteTokens.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    setAmountUSD: setToAmountUSD,
+    setAmount: setToAmount,
+    token: toToken,
+    loading: toLoading,
+    setLoading: setToLoading,
+    error: toTokenBalanceError,
+  });
+
+  return { fromETH, from, fromUSDC, to };
+};
```
Comment (alessey) [inline src/swap/hooks/useSwapLiteTokens.ts:104]: is it possible to extract a single useToken hook and use it for each usecase?

File: src/swap/constants.ts [hunk 4]
```diff
@@ -23,3 +26,23 @@ export enum SwapMessage {
...
+
+export const usdcToken: Token = {
+  name: 'USDC',
+  address: '0x833589fcd6edb6e08f4c7c32d4f71b54bda02913',
+  symbol: 'USDC',
+  decimals: 6,
+  image:
+    'https://d3r81g40ycuhqg.cloudfront.net/wallet/wais/44/2b/442b80bd16af0c0d9b22e03a16753823fe826e5bfd457292b55fa0ba8c1ba213-ZWUzYjJmZGUtMDYxNy00NDcyLTg0NjQtMWI4OGEwYjBiODE2',
+  chainId: base.id,
+};
```
Comment (0xAlec) [inline src/swap/constants.ts:48]: should these belong in https://github.com/coinbase/onchainkit/tree/main/src/token instead?

File: src/swap/hooks/useSwapLiteTokens.ts (hunk 3 above)
Comment (0xAlec) [reply src/swap/hooks/useSwapLiteTokens.ts:104]: +1, I think we should use a shared util for each input

File: src/swap/constants.ts (hunk 4 above)
Comment (abcrane123) [reply src/swap/constants.ts:48]: yeah i'll move them

File: src/swap/hooks/useSwapLiteTokens.ts (hunk 3 above)
Comment (abcrane123) [reply src/swap/hooks/useSwapLiteTokens.ts:46]: done ! let me know what you think

File: src/swap/hooks/useResetSwapLiteInputs.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      fromETH.setAmountUSD(''),
+      fromUSDC.balanceResponse?.refetch(),
+      fromUSDC.setAmount(''),
+      fromUSDC.setAmountUSD(''),
+      to.balanceResponse?.refetch(),
+      to.setAmount(''),
+      to.setAmountUSD(''),
+    ]);
+  }, [from, fromETH, fromUSDC, to]);
+};
```
Comment (alessey) [inline src/swap/hooks/useResetSwapLiteInputs.ts:27]: this may be more flexible accepting an array of SwapUnits

**Review by alessey — APPROVED** [review]

File: src/swap/hooks/useResetSwapLiteInputs.ts (hunk 5 above)
Comment (abcrane123) [reply src/swap/hooks/useResetSwapLiteInputs.ts:27]: created a ticket to refactor later

**Pull request merged**